cmd/cloudshell_open/cloudrun.go

```diff
@@ -112,14 +113,24 @@ func tryFixServiceName(name string) string {
 	if name == "" {
 		return name
 	}
+
+	name = strings.ToLower(name)
+
 	if !unicode.IsLetter([]rune(name)[0]) {
 		name = fmt.Sprintf("svc-%s", name)
 	}
+
 	if len(name) > 63 {
 		name = name[:63]
 	}
+
 	for name[len(name)-1] == '-' {
 		name = name[:len(name)-1]
 	}
+
+	reg, _ := regexp.Compile("[^a-z0-9-]+")
```

Comment:
use regexp.MustCompile so we crash the process in case regexp is incorrect.
ideally ignoring errors is rarely a good thing in Go.

Reply:
you can also directly do MustCompile(..).ReplaceAllString(...)

Reply:
Thanks! As you can tell, I'm new to this Go thing :)